File: 2023/advent_of_code/test/q15/q15_test.exs
defmodule AdventOfCode.Q15Test do
  use ExUnit.Case
  doctest AdventOfCode.Q15

  test "Q15 hash works" do
    result = AdventOfCode.Q15.hash("HASH" |> String.graphemes())
    assert result == 52
  end

  test "Q15 Part 1 Example works" do
    file_stream = File.stream!("./test/q15/q15_example.txt")
    result = AdventOfCode.Q15.part1(file_stream)
    assert result == 1320
  end

  test "Q15 Part 1 works" do
    file_stream = File.stream!("./test/q15/q15.txt")
    result = AdventOfCode.Q15.part1(file_stream)
    assert result == 509_152
  end

  test "Q15 Part 2 Example works" do
    file_stream = File.stream!("./test/q15/q15_example.txt")
    result = AdventOfCode.Q15.part2(file_stream)
    assert result == 145
  end

  test "Q15 Part 2 works" do
    file_stream = File.stream!("./test/q15/q15.txt")
    result = AdventOfCode.Q15.part2(file_stream)
    assert result == 244_403
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q15/Q15Test.kt
package q15

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q15Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q15.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(4919281, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q15.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(12630143363767, part2(filePath!!))
    }
}